addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,274 @@
+ *
+ * @author CaptainFreak
+ * @author ZainabAlShowely
+ */
+public class XSLTInjection extends AbstractAppParamPlugin {
+    private static final String MESSAGE_PREFIX = "ascanalpha.xsltinjection.";
+
+    private enum XSLTInjectionCheck {
+        ERROR,
+        VENDOR,
+        PORTSCAN,
+        COMMAND_EXEC
+    }
+
+    private static String[] errorCausingPayloads = {"<"};
+
+    private static String[] evidenceError = {
+        "compilation error", "XSLT compile error", "SAXParseException"
+    };
+
+    private static String[] vendorReturningPayloads = {
+        "<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+        "system-property(\'xsl:vendor\')/>",
+        "\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+        "<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+    };
+
+    private static String[] evidenceVendors = {
+        "libxslt", "Microsoft", "Saxonica", "Apache", "Xalan", "SAXON", "Transformiix"
+    };
+
+    private static String[] evidencePortScanning = {
+        "failed to open stream",
+        "Invalid Http response",
+        "Connection Refused",
+        "No connection could be made because the target machine actively refused it",
+        "Can not load requested doc"
+    };
+
+    private static String[] cmdExecPayloads = {
+        "<xsl:variable name=\"rtobject\" select=\"runtime:getRuntime()\"/>\n"
+                + "<xsl:variable name=\"process\" select=\"runtime:exec($rtobject,'erroneous_command')\"/>\n"
+                + "<xsl:variable name=\"waiting\" select=\"process:waitFor($process)\"/>\n"
+                + "<xsl:value-of select=\"$process\"/>",
+        "<xsl:value-of select=\"php:function('exec','erroneous_command 2>&amp;1')\"/>"
+    };
+
+    private static String[] evidenceCmdExec = {
+        "Cannot run program", "erroneous_command: not found",
+    };
+
+    private static final Logger LOG = Logger.getLogger(XSLTInjection.class);
+
+    // used to check against the attack strength
+    private int requestsSent = 0;
+    // map setting the limits associated with each attack strength
+    private static Map<AttackStrength, Integer> strengthToRequestCountMap = new HashMap<>();
+
+    // Uses maps for checks to avoid having to pass
+    // the payloads, responses and resources strings as parameters
+    private static Map<XSLTInjectionCheck, String[]> payloads = new HashMap<>();
+    private static Map<XSLTInjectionCheck, String[]> responses = new HashMap<>();
+    private static Map<XSLTInjectionCheck, String> resourceIdentifiers = new HashMap<>();
+
+    static {
+        strengthToRequestCountMap.put(AttackStrength.DEFAULT, 12);
+        strengthToRequestCountMap.put(AttackStrength.LOW, 6);
+        strengthToRequestCountMap.put(AttackStrength.MEDIUM, 12);
+        strengthToRequestCountMap.put(AttackStrength.HIGH, 24);
+        strengthToRequestCountMap.put(AttackStrength.INSANE, 500);
```

Comment:
While I agree that 500 would be ridiculous that isn't what's happening here. Yes it would be best to know what the actual upper bound is, if that's easily doable @ZainabAlShowely. If not I think we can leave it as 500 for now and note that it needs to be addressed in the future. The current ascan alpha doesn't even have test components yet :grin: